otherlibs/unix/unix_unix.ml

```diff
@@ -962,13 +965,13 @@ end = struct
     | Some mutex -> Mutex.protect mutex f
 end
 
-let popen_mutex = lazy (Mutex.create ())
+let popen_mutex = Mutex.create ()
```

Comment:
I removed the `Lazy` wrapper from here.  A mutex should not be very expensive to allocate.

Reply:
The lazy wrapper is needed because of a restriction under our runtime4 that systhreads must be initialized prior to using `Mutex`.

Reply:
Hmm… Good point, but I think in this case a real mutex is only created on runtime5. Does that make a difference?

Reply:
The code as-is requires the lazy to be there, I don't think it will work correctly otherwise.

Reply:
See #3698